### PR TITLE
Display courses/instructor list first based on which is more relevant 

### PR DESCRIPTION
**Pull request by alex-shen1**
## What I did
Elasticsearch responses give you a `score` variable for each entry it returns, representing how closely it thinks that the given entry applies to the query. I keep track of the highest score for both the returned instructors and courses list (I'm *pretty* sure ES returns search result in order of score, so picking out the score for the first element of the responses should give the highest), and then decide which to display first based on which score is higher.

## Other thoughts
- I'm think there's probably a smarter way to handle the display logic in the `search.html` template, but that's the only way I could come up with.
- Not sure if it's necessary to have `decide_order()` as its own function when it's equivalent to just a single boolean evaluation, but I made it so `search()` wasn't cluttered.
- This isn't really significant, but the name of the branch is a complete misnomer because this was originally just a branch I made to play around with ES testing and I forgot to name a new branch when I started doing this.

### CODE REVIEW

**Comment by alex-shen1** [comment]
## Example
I'm not sure how to write actual Django tests for ES, but here are some (very limited) examples of usage.
image

image

However, it isn't always perfect (probably in this case because *field* matches up better; could probably either make the rule category average or just improving the algorithms)
image

Also, in case you're wondering, I haven't changed the background color; I just have a screen darkener that affects screenshots.

**Comment by j-alicia-long** [comment]
Is search case-sensitive at all?

File: tcf_website/views/search.py [hunk 1]
```diff
@@ -159,13 +166,19 @@ def format_instructors(results):
...
     if not courses["error"]:
         args["courses"] = courses["results"]
     if not instructors["error"]:
         args["instructors"] = instructors["results"]
     return args
+
+
+def decide_order(courses, instructors):
+    """Determines whether courses or instructors should be displayed first in search page."""
+    return courses["highest_score"] > instructors["highest_score"]
```
Comment (bradknaysi) [inline tcf_website/views/search.py:184]: ## Problem
I don't think comparing the highest scores is a safe way to decide ordering. Remember that each dataset is ranked by a DIFFERENT ranking algorithm, which means the scores are relative only to that specific algorithm and dataset. You need to normalize each set of scores before you compare them. 

## Solution
A simple solution would be to compute the z-score of the top result of each list. However, this makes the assumption that our course and instructor data is normally distributed (which is a fairly large assumption to make), but probably okay to make. You might want to do some research on normalization. I'm sure there better solutions out there. But, my intuition tells me that using z-score would be simple and effective.

## Pseudo-code for a z-score driven `decide_order`
```
def format_response(response):
      ...
      x_0 = results[0].get("_meta").get("score")
      μ = compute_mean_score(results)
      σ = compute_score_standard_deviation(results)
      formatted["z_0"] = (x_0 - μ) / σ
```
```
def decide_order(courses, instructors):
      return courses["z_0"] > instructors["z_0"]
```

File: tcf_website/views/review.py [hunk 2]
```diff
@@ -76,8 +76,12 @@ def new_review(request):
...
                 )
 
-                messages.add_message(request, messages.SUCCESS, 'Successfully reviewed '
-                                     + str(course_code) + '!')
+                messages.add_message(
+                    request,
+                    messages.SUCCESS,
+                    'Successfully reviewed ' +
+                    str(course_code) +
+                    '!')
```
[the remark this answers is not in the document]
Comment (alex-shen1) [reply tcf_website/views/review.py:84]: This change was just from me running `./precommit`; I didn't realize it was made until after I pushed.

**Comment by alex-shen1** [comment]
> Is search case-sensitive at all?

My current understanding is that ES is non-case-sensitive by default. This might not be the best way of proving that, but I searched `Bloomfield` and then `bloomfield` and got the same high score for both queries, so I would assume this is the case for our current implementation.

**Pull request closed without merging**